native/src/routes/Pois.tsx

```diff
@@ -135,7 +136,7 @@ const Pois = ({ pois: allPois, language, cityModel, route, navigation }: PoisPro
       deselectAll()
     }
   }
-  useOnBackNavigation(slug || multipoi ? deselectFeature : undefined)
+  useOnBackNavigation((slug || multipoi) && enteredViaMaps ? deselectFeature : undefined)
```

Comment:
this check looks weird. first you check if the slug or multipoi is there and then you check if `slug === undefined` (enteredViaMaps)